### PR TITLE
Switch OnDemand to Dex for authentication

### PR DESCRIPTION
**Pull request by treydock**
Update to Ondemand 1.8 through 'latest' repo
The port used by OnDemand must be the same inside and outside the container to keep configs simple.
Ensure CA cert is trusted for OnDemand as Dex does not seem to work with self-signed certs
Trust frontend SSH host key so that users going to frontend in shell app don't have to accept the key.

The login page:

Screen Shot 2020-07-07 at 7 40 22 AM

### CODE REVIEW

File: ondemand/install.sh [hunk 1]
```diff
@@ -10,50 +10,88 @@ log_info() {
...
-log_info "Configuring Ondemand for PAM based logins.."
-
-# This is to allow ondemand to use pam for local logins in docker:
-chmod 640 /etc/shadow
-chgrp apache /etc/shadow
-
-echo "LoadModule authnz_pam_module modules/mod_authnz_pam.so" > /opt/rh/httpd24/root/etc/httpd/conf.modules.d/55-authnz_pam.conf
-cp /usr/lib64/httpd/modules/mod_authnz_pam.so /opt/rh/httpd24/root/usr/lib64/httpd/modules/
-cp /etc/pam.d/sshd /etc/pam.d/ood-webapp
+log_info "Configuring Ondemand for LDAP based logins.."
```
Comment (plessbd) [inline ondemand/install.sh:25]: Is this still a valid comment?  Doesn't seem like it actually doing anything

Comment (plessbd) [reply ondemand/install.sh:25]: nevermind, just noticed the yml with the dex configuration.

Comment (treydock) [reply ondemand/install.sh:25]: Sort of?  In reality what we are doing is configuring OnDemand as we setup `ood_portal.yml` which is the core OnDemand config file for Apache and Dex.  The configuration of Dex is to talk to LDAP to the log message is accurate just maybe not specific enough?

Comment (treydock) [reply ondemand/install.sh]: Updated wording to be more accurate

**Pull request merged**